SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Utils/DefaultApplicationSettings.cs

```diff
@@ -51,33 +67,55 @@ public bool ContainsKey(string key)
         {
             lock (configLock)
             {
-                return configuration.AppSettings.Settings[key] != null;
+                if (configuration != null)
+                {
+                    return configuration.AppSettings.Settings[key] != null;
+                }
+                else
+                {
+                    AppCenterLog.Warn(AppCenterLog.LogTag, $"{CorruptedConfigurationWarning} Method 'ContainsKey', key:{key}");
+                }
             }
+            return false;
         }
 
         public void Remove(string key)
         {
             lock (configLock)
             {
-                configuration.AppSettings.Settings.Remove(key);
-                SaveConfiguration();
+                if (configuration != null)
+                {
+                    configuration.AppSettings.Settings.Remove(key);
+                    SaveConfiguration();
+                }
+                else
+                {
+                    AppCenterLog.Warn(AppCenterLog.LogTag, $"{CorruptedConfigurationWarning} Method 'Remove', key:{key}");
+                }
             }
         }
 
         private void SaveValue(string key, string value)
         {
             lock (configLock)
             {
-                var element = configuration.AppSettings.Settings[key];
-                if (element == null)
+                if (configuration != null)
                 {
-                    configuration.AppSettings.Settings.Add(key, value);
+                    var element = configuration.AppSettings.Settings[key];
```

Comment:
Will `configuration.AppSettings.Settings.Contains(key)` work here?

Reply:
In this code property 'Settings' is 'KeyValueConfigurationCollection' class object and this class has no method 'Contains'.